camel/embeddings/gemini_embedding.py

```diff
@@ -0,0 +1,115 @@
+# ========= Copyright 2023-2024 @ CAMEL-AI.org. All Rights Reserved. =========
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ========= Copyright 2023-2024 @ CAMEL-AI.org. All Rights Reserved. =========
+from __future__ import annotations
+
+import os
+from typing import Any
+
+from camel.embeddings.base import BaseEmbedding
+from camel.types import EmbeddingModelType, GeminiEmbeddingTaskType
+from camel.utils import api_keys_required
+
+
+class GeminiEmbedding(BaseEmbedding[str]):
+    r"""Provides text embedding functionalities using Google's Gemini models.
+
+    Args:
+        model_type (EmbeddingModelType, optional): The model type to be
+            used for text embeddings.
+            (default: :obj:`GEMINI_EMBEDDING_EXP`)
+        api_key (str, optional): The API key for authenticating with the
+            Gemini service. (default: :obj:`None`)
+        dimensions (int, optional): The text embedding output dimensions.
+            (default: :obj:`None`)
+        task_type (GeminiEmbeddingTaskType, optional): The task type for which
+            to optimize the embeddings. (default: :obj:`None`)
+
+    Raises:
+        RuntimeError: If an unsupported model type is specified.
+    """
+
+    @api_keys_required(
+        [
+            ("api_key", 'GEMINI_API_KEY'),
+        ]
+    )
+    def __init__(
+        self,
+        model_type: EmbeddingModelType = (
+            EmbeddingModelType.GEMINI_EMBEDDING_EXP
+        ),
+        api_key: str | None = None,
+        dimensions: int | None = None,
+        task_type: GeminiEmbeddingTaskType | None = None,
```

Comment:
use Optional to align with camel's code style

```suggestion
        api_key: Optional[str] = None,
        dimensions: Optional[int] = None,
        task_type: Optional[GeminiEmbeddingTaskType] = None,
```